docs/prover/checking/coverage-info.md

```diff
@@ -0,0 +1,147 @@
+
+
+
+###  Visualization on the `.sol` and `.spec` files
+
+You can access the unsat cores visualization via `Job Info -> Unsat Core Page` buttons, as shown on Figure 1. 
+
+![](coverage-info-button.png)
+Figure 1: The `.spec` & `.sol` visualization button.
+
+The visualization itself is shown on Figure 2. It consists of two *panes*: 
+1. The left pane shows *per-line* visualization.
+2. The right pane shows detailed info about individual visualized lines. 
+
+![](tautology-sol-and-spec-new.png)
+Figure 2: The visualization of the Tautology example on `.sol` and `.spec`. 
+
+
+In particular, in the left pane, we highlight lines from individual `.sol` and `.spec` files. 
+Every line can have either none, green, red or yellow background color. No background means we have no information about this line. For instance, it might not be neither `assert` nor `require` nor `assign` command, or we might not have the mapping due to compilation to the EVM bytecode. Green, red or yellow background means that we have a mapping to one or more commands on the line (one `.spec`/`.sol` line can be often broken down to several TAC commands) with the following meaning:
+
+1. Green means that all of the TAC commands that are mapped to the line are in the unsat core, i.e. needed to prove the property. 
+2. Red means that none of the TAC commands that are mapped to the line are in the unsat core, i.e. not needed to prove the property.
+3. Yellow means that some of the TAC commands that are mapped to the line are in the unsat core and some of them are not in the unsat core.
+
+Furthermore, if we have multiple rules/invariants or a parametric rule (such as our `tautology`), we can also have multiple rules/invariants mapping to a single `.sol` or `.spec` line. That is, we generate just a single, joint, visualization for all the rules/invariants (run with `--rule` and `--method` flags to get a visualization for a single rule/method/invariant). In such a case, a yellow line means that some of the commands on the line are needed to prove some of the rules/methods/invariants and some of the commands are not needed. 
+
+The right pane provides detailed information about individual lines and commands mapped to the lines group by the rule/method/invariant name (denoted *rule*) and the *value*, where each *value* refers to a particular `assign`, `assert` or `assume` TAC command that is mapped to the line. For instance, in the Tautology example, we can see that sub-command `other != manager` of line `11` matters (is in the unsat core), whereas `other != e.msg.sender` is not in the unsat core. Also, note that the pane shows e.g. both `other != manager` and `other == manager`; this is because in our TAC representation we encode `other != manager` as `!(other == manager)`. 
+
+
+### Visualization on TAC 
```

Comment:
Alex gave webinar just a few weeks ago about using our TAC dumps to better understand and resolve timeouts. No one is saying that the end users have to look at a TAC dump; but if they can, it can help them a lot. Moreover, there is even a button (for long time already) to the TAC dump from the HTML report. 
1. Perhaps we should open a broader discussion (including other folks from the company) about allowing/disallowing end users to work with TAC
2. If we do not want to include TAC in the official documentation "for basic users", is there a place with documentation for "advanced users"?

Reply:
Yeah, I figure we haven't synced on this enough...
My understanding, e.g. with Nurit and Shelly, lately was that we want to cautiously expose Security Engineers, and advanced end users to the graphical TAC dumps (not 100% sure about those groups, perhaps we can define it better).
Shelly has been making them nicer for that reason, and I have for example based some of the timeout feedback on them.

I think we're all in agreement that it would be nicer to lift everything user-facing to spec / source level, but this is, among other things, a developer-time tradeoff (not necessarily an aspect we advertise, but still..).
Another aspect is that TAC is closer to the VC, so for instance when it comes to understanding an SMT timeout, it might be helpful to see "what the solver is seeing" (e.g. when someone thinks they've summarized all nonlinear away, but haven't, they can see it in TAC).

The plan, afaiu, was to communicate a very rough working understanding about the TAC dumps. 
As of now, this mostly includes looking at
 - the control flow graph, including the external calls that can be seen clearly in our graphs,
 - the "back pointers" that, on hover over some tac statements appear and show source/spec locations.

We should probably have discussed / communicated this with you Mike in particular earlier -- I guess it didn't make the way across the ocean :-/ ...
(also not against a wider discussion, but I understand it's especially Mike's purview ..)
 --> happy to discuss in any combination

Reply:
Note also that I have an upcoming PR that adds a stub for that TAC dump documentation -- which will be followed up by a PR for that documentation itself ... (actually, I'll make a jira issue right now ..)

Reply:
removed